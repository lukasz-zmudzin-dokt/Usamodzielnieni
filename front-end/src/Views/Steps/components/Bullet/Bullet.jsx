import React from 'react';

const Bullet = ({ step }) => {
    return (
        <>
            <div className={`bullet bullet--${step.type}`}>
                {step.type === 'main' && step.title}
            </div>
            {step.type === 'sub' && <div className="bullet__title">{step.title}</div>}
        </>
    )
}

export default Bullet;
