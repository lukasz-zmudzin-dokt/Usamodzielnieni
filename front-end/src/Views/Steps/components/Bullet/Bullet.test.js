import React from "react";
import { render } from "@testing-library/react";
import Bullet from "./Bullet";

describe("Bullet", () => {
  let props;

  beforeEach(() => {
    props = {
      step: {
        type: "main",
        title: "TytuÅ‚ kroku",
      },
    };
  });

  it("should render without crashing", () => {
    const { container } = render(<Bullet {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing when step type is sub", () => {
    props.step.type = "sub";
    const { container } = render(<Bullet {...props} />);
    expect(container).toMatchSnapshot();
  });
});
