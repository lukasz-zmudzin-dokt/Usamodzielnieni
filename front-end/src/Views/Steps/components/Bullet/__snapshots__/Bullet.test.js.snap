// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet should render without crashing 1`] = `
<div>
  <div
    class="bullet bullet--main"
  >
    <span>
      Tytuł kroku
    </span>
  </div>
</div>
`;

exports[`Bullet should render without crashing when step type is sub 1`] = `
<div>
  <div
    class="bullet bullet--sub"
  />
  <div
    class="bullet__title"
  >
    <span>
      Tytuł kroku
    </span>
  </div>
</div>
`;
